test/env/lib.rs

```diff
@@ -147,8 +148,35 @@ impl Transfer {
     }
 }
 
+#[async_trait(?Send)]
+pub trait LedgerEnv {
+    fn fork(&self) -> Self;
+    fn principal(&self) -> Principal;
+    async fn query<Input, Output>(&self, method: &str, input: Input) -> anyhow::Result<Output>
+    where
+        Input: ArgumentEncoder + std::fmt::Debug,
+        Output: for<'a> ArgumentDecoder<'a>;
+    async fn update<Input, Output>(&self, method: &str, input: Input) -> anyhow::Result<Output>
+    where
+        Input: ArgumentEncoder + std::fmt::Debug,
+        Output: for<'a> ArgumentDecoder<'a>;
+}
+
+#[async_trait(?Send)]
+pub trait LedgerEndpoint: LedgerEnv {
+    async fn transfer(&self, arg: Transfer) -> anyhow::Result<Result<Nat, TransferError>>;
+    async fn balance_of(&self, account: impl Into<Account>) -> anyhow::Result<Nat>;
+    async fn supported_standards(&self) -> anyhow::Result<Vec<SupportedStandard>>;
+    async fn metadata(&self) -> anyhow::Result<Vec<(String, Value)>>;
+    async fn minting_account(&self) -> anyhow::Result<Option<Account>>;
+    async fn token_name(&self) -> anyhow::Result<String>;
+    async fn token_symbol(&self) -> anyhow::Result<String>;
+    async fn token_decimals(&self) -> anyhow::Result<u8>;
+    async fn transfer_fee(&self) -> anyhow::Result<Nat>;
```

Comment:
There should be only one "right" implementation for all of these methods in terms of LedgerEnv::query and LedgerEnv::update, so let's provide the default implementation for all methods.
This way, we won't need to repeat the exact implementation for all instances of LedgerEnv.